File: 119-pascals-triangle-ii/pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> prev(2,1), cur(1,1);
        if (rowIndex == 0) return cur;
        else if (rowIndex == 1) return prev;
        for (int i=2; i<=rowIndex; i++){
            cur = {1};
            for (int j=0; j<i-1; j++){
                cur.push_back(prev[j]+prev[j+1]);
            }
            cur.push_back(1);
            prev = cur;
        }
        return prev;
    }
};

File: 119-pascals-triangle-ii/pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        if rowIndex == 0:
            return [1]
        elif rowIndex == 1:
            return [1,1]
        prev = [1,1]
        cur = []
        for i in range(2,rowIndex+1):
            cur = [1]
            for j in range(i-1):
                cur.append(prev[j]+prev[j+1])
            cur.append(1)
            prev,cur = cur,prev
        return prev